pageserver/src/virtual_file/owned_buffers_io/io_buf_ext.rs

```diff
@@ -0,0 +1,41 @@
+use bytes::{Bytes, BytesMut};
+use std::ops::Range;
+use tokio_epoll_uring::{BoundedBuf, Slice};
+
+pub(crate) trait IoBufExt {
+    /// Get a [`Slice`] covering the range `[0..self.len()]`.
+    /// It is guaranteed that the resulting slice has [`Slice::bytes_init`] equal to [`Slice::bytes_total`].
+    ///
+    /// This is for use on the write path.
+    fn slice_len(self) -> Slice<Self>
+    where
+        Self: Sized;
+}
+
+macro_rules! impl_io_buf_ext {
+    ($($t:ty),*) => {
+        $(
+            impl IoBufExt for $t {
+                #[inline(always)]
+                fn slice_len(self) -> Slice<Self> {
+                    let len = self.len();
+                    let s = if len == 0 {
+                        // paper over the incorrect assertion
+                        // https://github.com/neondatabase/tokio-epoll-uring/issues/46
```

Comment:
Can you link to the assertion? The issues doesn't have it.

Reply:
Updated the issue.